src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1529,6 +1529,10 @@ protected override void ProcessRecord()
                                         response.ReasonPhrase);
 
                                     HttpResponseException httpEx = new(message, response);
+                                    {
+                                        StatusCode = response.StatusCode;
```

Comment:
Eh, StatusCode is read-only property - we cannot assign it :-(. Only new constructor can initialize this.

Reply:
We can do this in current constructor:
```c#
        public HttpResponseException(string message, HttpResponseMessage response) : base(message, inner: null, response.StatusCode)
        {
            Response = response;
        }
```

Reply:
Done, very elegant solution